sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetOptions.scala

```diff
@@ -42,8 +43,15 @@ private[parquet] class ParquetOptions(
    * Acceptable values are defined in [[shortParquetCompressionCodecNames]].
    */
   val compressionCodecClassName: String = {
-    val codecName = parameters.getOrElse("compression",
-      sqlConf.parquetCompressionCodec).toLowerCase(Locale.ROOT)
+    // `compression`, `parquet.compression`(i.e., ParquetOutputFormat.COMPRESSION), and
+    // `spark.sql.parquet.compression.codec`
+    // are in order of precedence from highest to lowest.
+    val parquetCompressionConf = parameters.get(ParquetOutputFormat.COMPRESSION)
+    val codecName = parameters
+      .get("compression")
+      .orElse(parquetCompressionConf)
```

Comment:
Is this new? Do we support `parquet.compression` before this PR?

Reply:
Yes it's new. I guess `PartitionOptions` did not used when writing hive table before, because it's invisible for hive. I changeed it to `public`.

Reply:
Could we keep the old behavior? We could add it later? We do not want to mix multiple issues in the same PR?

Reply:
If so, parquet's table-level compression may be overwrited in this PR, and it may not be what we want.
Shall I  fix it first in another PR?

Reply:
Yeah, we can submit a separate PR for that issue. The behavior change needs to be documented in SparkSQL doc.